OpenSim/Common/DataTable.h

```diff
@@ -121,6 +121,163 @@ class DataTable_ : public AbstractDataTable {
+        // No operation.
+    }
+
+    /** Copy assign a DataTable_<double, double> from 
+    DataTable_<double, ThatETY> where ThatETY can be SimTK::Vec<X>. Each column
+    of the other table is split into multiple columns of this table. For example
+    , DataTable_<double, Vec3> with 3 columns and 4 rows will construct
+    DataTable<double, double> of 9 columns and 4 rows where each component of
+    SimTK::Vec3 ends up in one column. Column labels of the resulting DataTable
+    will use column labels of source table appended with suffixes of form "_1",
+    "_2", "_3" and so on.
+
+    \tparam ThatETY Datatype of the matrix underlying the given DataTable.
+
+    \param that DataTable to copy assign from. This table can be
+                of for example DataTable_<double, Quaternion>, 
+                DataTable_<double, Vec6>.
+
+    \throws InvalidArgument If 'that' DataTable has no column-labels.
+    \throws InvalidArgument If 'that' DataTable has zero number of rows/columns.
+    \throws InvalidArgument If 'suffixes' does not contain same number of
+                            elements as that.numComponentsPerElement().       */
+    template<typename ThatETY>
+    DataTable_& operator=(const DataTable_<double, ThatETY>& that) {
+        return operator=(DataTable_{that});
+    }
+
+    /** Flatten the columns of this table to create a 
+    DataTable_<double, double>. The each column will be split into its 
+    constituent components. For example, each column of a 
+    DataTable_<double, Vec3> will be split into 3 columns. The column-labels of
+    the resulting columns will be suffixed "_1", "_2", "_3" and so on. See
+    documentation for constructor DataTable_::DataTable_().                   */
+    DataTable_<double, double> flatten() const {
+        return DataTable_<double, double>{*this};
+    }
+
+    /** Flatten the columns of this table to create a 
+    DataTable_<double, double>. The each column will be split into its 
+    constituent components. For example, each column of a 
+    DataTable_<double, Vec3> will be split into 3 columns. The column-labels of
+    the resulting columns will be appended with 'suffixes' provided. See
+    documentation for constructor DataTable_::DataTable_().                   */
+    DataTable_<double, double>
+    flatten(const std::vector<std::string>& suffixes) const {
+        return DataTable_<double, double>{*this, suffixes};
+    }
+
+    /** Retrieve the number of components each element (of type ETY) of the 
+    table is made of. Some examples:
+    <table>
+    <tr><td>Table Type</td><td>Element Type</td><td>Num of Components</td></tr>
+    <tr><td>DataTable<double, double></td><td>double</td><td>1</td></tr>
+    <tr><td>DataTable<double, Vec3></td><td>Vec3</td><td>3</td></tr>
+    <tr><td>DataTable<double, Quaternion></td><td>Quaternion</td><td>4</td></tr>
+    </table>                                                                  */
```

Comment:
I have changed the table to use markdown. It is much more readable now in the code.